src/index.tsx

```diff
@@ -43,10 +54,18 @@ function useInputFile({ ref, options }: Props) {
         setFiles(files);
       };
 
-      input.addEventListener('change', getFiles, false);
+      const eventListener = (event: Event) => {
+        if (customerHandler.current) {
+          return customerHandler.current(event);
+        }
+
+        return getFiles(event);
+      };
+
+      input.addEventListener('change', eventListener, false);
 
       return () => {
-        input.removeEventListener('change', getFiles, false);
+        input.removeEventListener('change', eventListener, false);
```

Comment:
double-check in here, I think it's removing event listener each time.

Reply:
I log on here and confirmed not removing event listener each time.

Reply:
that's strange then...